src/hooks/use-touched-state.tsx

```diff
@@ -0,0 +1,29 @@
+import { useState, FocusEvent, useCallback } from "react"
+
+type BlurHandler = (e: FocusEvent) => void
+
+type UseTouchedStateHookValues = [
+  boolean,
+  BlurHandler,
+  React.Dispatch<React.SetStateAction<boolean>>
+]
+
+export const useTouchedState = (
```

Comment:
I think I will move it under the input directory, seems to be not generic enough.